python/nammu/utils/__init__.py

```diff
@@ -271,12 +271,15 @@ def get_config_versions(path_to_jar, yaml_path, path_to_config):
     try:
         jar_contents = zipfile.ZipFile(path_to_jar, 'r')
     except zipfile.BadZipfile:
-        jar_config = yaml.safe_load(open(path_to_jar, 'r'))
+        with open(path_to_jar, 'r') as jar:
+            jar_config = yaml.safe_load(jar)
     else:
-        jar_config = yaml.safe_load(jar_contents.open(yaml_path))
+        with jar_contents.open(yaml_path) as jar:
+            jar_config = yaml.safe_load(jar)
```

Comment:
Actually, this still doesn't close the jar file (if it was opened successfully). `ZipFile` [can also be used as a context manager](https://docs.python.org/2/library/zipfile.html#zipfile-objects), so the following structure might be slightly better:

```python
try:
    with zipfile.ZipFile(path_to_jar, 'r') as jar_contents:
        with jar_contents.open(yaml_path) as jar:
            jar_config = yaml.safe_load(jar)
except zipfile.BadZipfile:
    with open(path_to_jar, 'r') as jar:
        jar_config = yaml.safe_load(jar)
```

If this is less readable, we can instead add  a `jar_contents.close()` at the end of the `else` caluse.